crypto/secp256k1.go

```diff
@@ -6,7 +6,8 @@ import (
 
 	pb "github.com/libp2p/go-libp2p-core/crypto/pb"
 
-	"github.com/btcsuite/btcd/btcec"
+	btcec "github.com/btcsuite/btcd/btcec/v2"
+	ecdsa "github.com/btcsuite/btcd/btcec/v2/ecdsa"
```

Comment:
rename to btcecdsa (there's an ecdsa in the stdlib).

Reply:
> Needs a rebase, but otherwise this looks fine. It's technically a breaking change because `Secp246k1{Public,Private}Key` is a newtype (and we're changing the underlying type). However, I don't _personally_ consider that to be a part of the API.

I agree that this didn't feel like part of the public interface. I don't know how to reset this to be the new, accepted interface. Can you point me in the right direction to get that test to pass.

Reply:
@marten-seemann has the final decision here.

Reply:
Ah, thanks.